### PR TITLE
Add Apluxs Pulsar XBOARD QS keyboard

### PR DESCRIPTION
**Pull request by skylupin03**
## Description

This is a PR to add the keyboard "Aplusx, PULSAR XBOARD QS" to QMK.

Thank you for taking the time to review this submission 
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/aplusx_pulsar/xboard_qs/config.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define WAIT_US_TIMER           GPTD3       //  Encoder 및 RGB ANI가 부드러워짐
+
+// Dynamic EEPROM
+#define DYNAMIC_KEYMAP_EEPROM_MAX_ADDR  4095
+
```
Comment (zvecr) [inline keyboards/aplusx_pulsar/xboard_qs/config.h:23]:
```suggestion
```

File: keyboards/aplusx_pulsar/xboard_qs/info.json [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "encoder": true,
+        "rgb_matrix": true
+    },
+    "bootmagic": {
+        "matrix": [2, 7]
+    },
```
Comment (zvecr) [inline keyboards/aplusx_pulsar/xboard_qs/info.json:34]:
```suggestion
    },
    "eeprom": {
        "wear_leveling": {
            "backing_size": 4096
        }
    },
```

File: keyboards/aplusx_pulsar/xboard_qs/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        case KC_LGUI:
+            if (win_key_locked) { return false; }
+            else{ return true;}
+
+        default:
+            return true;   // Process all other keycodes normally
+    }
+    return true;
+}
```
Comment (zvecr) [inline keyboards/aplusx_pulsar/xboard_qs/keymaps/default/keymap.c:98]: This already exists as `QK_MAGIC_TOGGLE_GUI`

File: keyboards/aplusx_pulsar/xboard_qs/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+            if (record->event.pressed) {                // Toggle GUI lock on key press
+                win_key_locked = !win_key_locked;
+            }
+            break;
+
+        case KC_LGUI:
+            if (win_key_locked) { return false; }
+            else{ return true; }
+
+        case KC_MISSION_CONTROL:
```
Comment (zvecr) [inline keyboards/aplusx_pulsar/xboard_qs/keymaps/via/keymap.c:191]: Some of these already exist in code, and should not be reimplemented at the keymap level

File: keyboards/aplusx_pulsar/xboard_qs/xboard_qs.c
Comment (zvecr) [inline keyboards/aplusx_pulsar/xboard_qs/xboard_qs.c:1]: This file should be removed, and its content migrated to `rgb_matrix.layout` within info.json. `qmk info -f json -kb aplusx_pulsar/xboard_qs` should help with this.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/aplusx_pulsar/xboard_qs/xboard_qs.c
Comment (skylupin03) [reply keyboards/aplusx_pulsar/xboard_qs/xboard_qs.c:1]: Thank you very much for your precious time.
I learned as much as I could to modify everything

**Comment by skylupin03** [comment]
While learning while referring to other people's codes, I accidentally changed the code,
 I'm so sorry. I revised it as carefully as possible
I'm sorry I'm not good enough

**Comment by skylupin03** [comment]
It's gotten a lot complicated. I'll clean it up and do PR again, thank you for reviewing it

**Pull request closed without merging**